src/theory/strings/core_solver.cpp

```diff
@@ -1638,7 +1646,7 @@ void CoreSolver::processSimpleNEq(NormalForm& nfi,
             lentTestSuccess = e;
             lenConstraint = entLit;
             // its not explained by the equality engine of this class
-            iinfo.d_antn.push_back(lenConstraint);
+            iinfo.d_noExplain.push_back(lenConstraint);
```

Comment:
Don't we also need to push to `iinfo.d_ant`?

Reply:
It is being pushed into `d_ant`.  The length constraint is being accumulated into `lcVec`, which is added to `d_ant` on line 1711.  I'll add some further comments to make this clearer.

Reply:
Oh, I see. Thanks for the comment.